### PR TITLE
Add support for Mapy.cz map source

### PR DESCRIPTION
**Pull request by jirkavrba**
## Description
I added support for [Mapy.cz](https://mapy.cz) map source, mostly for me as my personal fork, but I think other users could benefit from this as well, as the map quality seems overall better in Europe in my opinion.

### CODE REVIEW

**Comment by Lineflyer** [comment]
ok to test

**Comment by bekuno** [comment]
Are the point license and map attribution solved?
See https://licence.mapy.cz/?doc=mapy_pu
On the web map it is attributed with:
image

**Comment by jirkavrba** [comment]
I'm pretty new to this topic, the `and others` leads to popup with OpenStreetMap contributors, is there a practical difference in displaying `and <a href=osm>others</a>` and `and <a href=osm>OpenStreetMap contributors</a>`?

**Comment by bekuno** [comment]
> 
> 
> I'm pretty new to this topic, the `and others` leads to a popup with OpenStreetMap contributors, is there a practical difference in displaying `and <a href=osm>others</a>` and `and <a href=osm>OpenStreetMap contributors</a>`?
> 
> I can add the correct version to the map attribution

I was not involved in the last implementation, but you should find some information in earlier Issues and PR, eg. https://github.com/cgeo/cgeo/issues/9411 and https://github.com/cgeo/cgeo/issues/6150.

**Comment by jirkavrba** [comment]
Thanks a lot, I will take a look at it

**Comment by jirkavrba** [comment]
From my understanding of the license, the attribution after commit https://github.com/cgeo/cgeo/pull/11419/commits/f6670ae65214e4efa09a2b936ada2b1bc431758b should be valid

File: main/res/values/strings_not_translatable.xml [hunk 1]
```diff
@@ -58,6 +59,7 @@
     <string translatable="false" name="map_attribution_openstreetmap_html"><![CDATA[<a href="https://www.openstreetmap.org/copyright">© OpenStreetMap contributors</a>]]></string>
     <string translatable="false" name="map_attribution_openstreetmapde_html"><![CDATA[© <a href="https://openstreetmap.de/">OpenStreetMap DE</a>, map data <a href="https://www.openstreetmap.org/copyright">OpenStreetMap contributors</a>]]></string>
     <string translatable="false" name="map_attribution_cyclosm_html"><![CDATA[© <a href="https://www.cyclosm.org/">CyclOSM</a>, map data <a href="https://www.openstreetmap.org/copyright">OpenStreetMap contributors</a>]]></string>
+    <string translatable="false" name="map_attribution_mapy_cz"><![CDATA[© <a href="https://o.seznam.cz/">Seznam.cz, a.s.</a> and the <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors]]></string>
```
[the remark this answers is not in the document]
Comment (moving-bits) [reply main/res/values/strings_not_translatable.xml:62]: and please add an attribution to `values/strings_not_translatable.xml` in the `components` var, see "More maps based on ..." near the end. This text will be shown in the "about c:geo" section of c:geo, wheres the string above is for display on the map.

**Review by moving-bits — COMMENTED** [review]
Do you know whether it's legally ok for c:geo to include this mapsource? (The license terms on mapy.cz were not really understandable for me. Is what we are trying to do here an "API usage"? etc.)

File: main/res/values/strings_not_translatable.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jirkavrba) [reply main/res/values/strings_not_translatable.xml:62]: It should be legal to include the map source, as long as it is mentioned in the attribution, given the fact that c:geo is not a commercial application. I'm not 100% sure tho...

**Comment by jirkavrba** [comment]
> Do you know whether it's legally ok for c:geo to include this mapsource? (The license terms on mapy.cz were not really understandable for me. Is what we are trying to do here an "API usage"? etc.)

It should be legal to include the map source, as long as it is mentioned in the attribution, given the fact that c:geo is not a commercial application. I'm not 100% sure tho...

File: main/res/values/strings_not_translatable.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jirkavrba) [reply main/res/values/strings_not_translatable.xml]: Fixed in commits https://github.com/cgeo/cgeo/pull/11419/commits/73d1dfa66776f4f8feb8793913aafd9e616bf49c and https://github.com/cgeo/cgeo/pull/11419/commits/fce2534464acb74fa2fe0a7ad9f660404e79f386

**Comment by moving-bits** [comment]
Thanks for updating the PR and thanks @jirkavrba for this contribution in general!

**Pull request merged**